src/main/features/response/routes/defendant-link-receiver.ts

```diff
@@ -7,36 +7,40 @@ import { Paths } from 'response/paths'
 
 import ClaimStoreClient from 'app/claims/claimStoreClient'
 import Claim from 'app/claims/models/claim'
+import User from 'idam/user'
+import { ErrorHandling } from 'common/errorHandling'
 
 const logger = require('@hmcts/nodejs-logging').getLogger('router/defendantLinkReceiver')
 
 export default express.Router()
-  .get(Paths.defendantLinkReceiver.uri, async (req: express.Request, res: express.Response, next: express.NextFunction) => {
-    const { letterHolderId } = req.params
-
-    if (!res.locals.user.isInRoles(`letter-${letterHolderId}`)) {
-      logger.error('User not in letter ID role - redirecting to access denied page')
-      res.redirect(ErrorPaths.claimSummaryAccessDeniedPage.uri)
-      return
-    }
-
-    try {
-      const claim: Claim = await ClaimStoreClient.retrieveByLetterHolderId(letterHolderId)
-
-      if (!claim.defendantId) {
-        try {
-          await ClaimStoreClient.linkDefendant(claim.id, res.locals.user.id)
-        } catch (err) {
-          next(err)
-          return
-        }
+  .get(Paths.defendantLinkReceiver.uri,
+    ErrorHandling.apply(async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+      const { letterHolderId } = req.params
+
+      const user: User = res.locals.user
+      if (!user.isInRoles(`letter-${letterHolderId}`)) {
+        logger.error('User not in letter ID role - redirecting to access denied page')
+        res.redirect(ErrorPaths.claimSummaryAccessDeniedPage.uri)
+        return
       }
 
-      const sessionCookie = config.get<string>('session.cookieName')
-      new Cookies(req, res).set(sessionCookie, req.query.jwt, { sameSite: 'lax' })
+      try {
+        const claim: Claim = await ClaimStoreClient.retrieveByLetterHolderId(letterHolderId)
+
+        if (!claim.defendantId) {
+          try {
+            await ClaimStoreClient.linkDefendant(claim.id, user.id)
+          } catch (err) {
+            next(err)
+            return
+          }
+        }
 
-      res.redirect(Paths.taskListPage.uri)
-    } catch (err) {
-      next(err)
-    }
-  })
+        const sessionCookie = config.get<string>('session.cookieName')
+        new Cookies(req, res).set(sessionCookie, req.query.jwt, { sameSite: 'lax' })
```

Comment:
Building this cookie looks like something that could be extracted to a function/class.